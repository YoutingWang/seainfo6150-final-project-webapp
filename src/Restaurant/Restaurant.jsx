import React from 'react'
import styles from "./Restaurant.module.css"

const Restaurant = (props) => {
    const business = props.business;
    if (!business) {
        return <div>You have no data!</div>;
    }

    const tags = business.categories.map(category => (<span className={styles.tag} key={business.id + category.title}>{category.title}</span>));
    const addressLines = business.location.display_address.map(addressLine => <p key={business.id + addressLine}>{addressLine}</p>);

    return (
        <div className={styles.result}>
            <img className={styles.img} src={business.image_url} alt='image'/>
            <div className={styles.info}>
                <h2 className="subtitle">{business.name}</h2>
                <p>{business.price} {tags}</p>
            </div>
            <div className={styles.contact}>
                <p>{business.phone}</p>
                {addressLines}
            </div>
        </div>
    )
}

export default Restaurant;