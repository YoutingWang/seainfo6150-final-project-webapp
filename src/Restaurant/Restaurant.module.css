.result {
    display: flex;
    max-width: 1135px;
    width: 100%;
    font-family: 'Raleway', sans-serif;
}
  
.img{
    display: block;
    width: 265px;
    height: 260px;
    border: 1px gray;
    margin-right: 1rem;
    margin-top: 16px;
    border-radius: 40px;
}
  
.info {
    min-width: 400px;
    font-family: 'Nunito', sans-serif;
}
  
.contact {
    font-size: 0.85rem;
    font-family: 'Nunito', sans-serif;
}
  
.tag {
    margin-right: 4px;
    font-family: 'Nunito', sans-serif;
}
  