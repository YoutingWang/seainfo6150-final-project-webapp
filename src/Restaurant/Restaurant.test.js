import React from "react";
import { render } from "@testing-library/react";
import Restaurant from "./Restaurant";
const restaurant = {
            "id": "ufl25yYz8lXeKzt11nZOUQ",
            "alias": "maldonados-pizzeria-mountain-view",
            "name": "Maldonado's Pizzeria",
            "image_url": "https://s3-media3.fl.yelpcdn.com/bphoto/V0ButCWCkna7ieqkq4dlRA/o.jpg",
            "is_closed": false,
            "url": "https://www.yelp.com/biz/maldonados-pizzeria-mountain-view?adjust_creative=z8902j_QQLYWdUdGPKjP3A&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=z8902j_QQLYWdUdGPKjP3A",
            "review_count": 526,
            "categories": [
                {
                    "alias": "pizza",
                    "title": "Pizza"
                }
            ],
            "rating": 4.0,
            "coordinates": {
                "latitude": 37.3963847,
                "longitude": -122.1009097
            },
            "transactions": [
                "delivery"
            ],
            "price": "$",
            "location": {
                "address1": "615 S Rengstorff Ave",
                "address2": "",
                "address3": "",
                "city": "Mountain View",
                "zip_code": "94040",
                "country": "US",
                "state": "CA",
                "display_address": [
                    "615 S Rengstorff Ave",
                    "Mountain View, CA 94040"
                ]
            },
            "phone": "+16509628499",
            "display_phone": "(650) 962-8499",
            "distance": 3389.2434837571523
};

describe("Restaurant tests", () => {
    it("renders correctly", () => {
        const { container } = render(<Restaurant restaurant={restaurant}/>);
        expect(container).toMatchSnapshot();
    });
}); 