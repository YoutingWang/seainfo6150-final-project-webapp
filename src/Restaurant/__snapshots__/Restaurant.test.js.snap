// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant tests renders correctly 1`] = `
<div>
  <div>
    You have no data!
  </div>
</div>
`;
